quesma/model/bucket_aggregations/date_histogram.go

```diff
@@ -286,17 +307,80 @@ func (qt *DateHistogramRowsTransformer) Transform(ctx context.Context, rowsFromD
 				i-1, rowsFromDB[i-1], i, rowsFromDB[i],
 			)
 		}
-		lastKey := qt.getKey(rowsFromDB[i-1])
-		currentKey := qt.getKey(rowsFromDB[i])
+		lastKey := qt.dateHistogram.calculateResponseKeyInUTC(qt.getKey(rowsFromDB[i-1]))
+		currentKey := qt.dateHistogram.calculateResponseKeyInUTC(qt.getKey(rowsFromDB[i]))
+
+		// ugly, but works, will do for now
+		doWeDivide := (currentKey/qt.getKey(rowsFromDB[i])) >= 100 || (float64(currentKey)/float64(qt.getKey(rowsFromDB[i]))) <= 0.01
```

Comment:
This check means the value is almost changed. Not sure if the division is right.


```suggestion
                keyFromDb := qt.getKey(rowsFromDB[i])
		doWeDivide := (currentKey/keyFromDb) >= 100 || (float64(currentKey)/float64(keyFromDb)) <= 0.01
```